### ISSUE TITLE
404 Client Error

### ISSUE THREAD

**Issue by jenbruno** [DCAN-Labs/nda-abcd-s3-downloader#81]
### A short summary of a question you have for the downloader development team.

Hi, I am getting the following error
requests.exceptions.HTTPError: 404 Client Error:  for url: https://nda.nih.gov/api/package/1228381/files

I have double checked that the package on NDA is current (created a few days ago) and keyring and passcode appear to be correctly set as well. I also searched for any previously downloaded files and could not find any. Any suggestions for a resolution?

Thank you!

**Comment by rosemccollum** [comment]
This error normally indicates an issue with the package. Although the package hasn't been archived, we have seen "current" packages fail to function due to something on the NDA's side. I would try to recreate your package and try again.

**Comment by jenbruno** [comment]
Thank you for your quick response. I have tried recreating the package and still am getting the 404 client error. I am trying a simpler package with only processed data and waiting for that to be ready for download. Do you have any other trouble shooting suggestions?

**Comment by rosemccollum** [comment]
Since the 404 error indicates something wrong with the package, I would reach out to the NDA and make sure there is nothing wrong with your packages and you haven't reached your download limit.

**Comment by eschettini18** [comment]
@jenbruno are you getting this error when running the download.py or downloadcmd command? I was getting that error when trying to run the download.py command for a new package I recently created but was following the instructions from prior to nda-tools version 0.2.25. When going back to the GitHub page today to try and identify my problem, and if I'm not interpreting the README page incorrectly (so please correct this if I'm wrong @rosemccollum), the download.py command does not currently work due to the nda-tools 0.2.25 security, so that could explain why you're having trouble accessing the package (if you did the same thing I did).

On a similar note... @rosemccollum, given that the README still indicates to use the downloadcmd, is it possible to only download a subset of subjects with this command, or would that have to wait until the s3 downloader upgrade is released? Trying to figure out the most efficient way to download all of the derivatives necessary to use your team's tfmri pipeline (https://github.com/DCAN-Labs/abcd-bids-tfmri-pipeline) without downloading subjects I won't be using in my analyses.

**Comment by rosemccollum** [comment]
@eschettini18 Yes it is still possible to only download a subset of subjects with the downloadcmd command. You can use the` --file-regex` flag to download files that match a specific pattern. If you run `downloadcmd -h`, it will show you examples of how to use this flag and others. You can also customize the s3 links file to only contain the files you want and provide it as an argument with the `-t` flag.

**Comment by JD-Hwang** [comment]
I guess this issue is same with my issue here.
Is there any way to download derivatives from DCAN labs?

https://github.com/ABCD-STUDY/nda-abcd-s3-downloader/issues/9#issue-2392018303

**Comment by eschettini18** [comment]
@JD-Hwang I've had success using the Interim Download Method that is outlined on this README page; of note, this page seems to be where they post upgrades for the s3-downloader rather than the ABCD-STUDY page you linked, which will be more relevant once they make it compatible with NDA version 0.2.27. I've been using the --file-regex flag Rose suggested for downloading subsets of subjects from packages I created from the 3165 project.

**Comment by JD-Hwang** [comment]
@eschettini18 Thank you for your comment! I was able to download the data package from NDA, but couldn't get nifti file that I wanted.
![image](https://github.com/DCAN-Labs/nda-abcd-s3-downloader/assets/65854964/2f18e887-1f2b-455a-8c15-836df9c96727)

I thought I could get it by running download.py with "datastructure_manifest.txt" from data package folder and 'data_subsets.txt", but It was not. Any clue what am I missing?

**Comment by JD-Hwang** [comment]
@eschettini18  Thanks a lot! I tried several times, and I figured out that I need to check the checkbox to include associated files (including nii files!) when I make data package haha

**Comment by victoris93** [comment]
@eschettini18 would you be willing to share the formatting of the file name that you have after `--file-regex`? I'm facing the issue of all nans in the "derived_files" column from `fmriresults01.txt` downloaded as part of the collection package. I assumed that one could just select the necessary derivative file paths from this column. Although the size of the package is 8GB which doesn't look right, is it the reason for the nans? At the point of adding items to the package I didn't unselect anything just to be sure to have everything. These are all the files that were included in the package:
```
md5_values.txt
imagingcollection01.txt
fmriresults01.txt
dataset_collection.txt
package_info.txt
package_file_metadata_1231549.txt.gz
README.pdf
```
